source/Cosmos.System2/FileSystem/FAT/Listing/FatDiretoryEntry.cs

```diff
@@ -787,7 +789,7 @@ internal void SetLongFilenameEntryMetadataValue(uint aEntryHeaderDataOffset, Fat
             if (xData.Length > 0)
             {
                 var xValue = new byte[aEntryMetadata.DataLength];
-                xValue = aValue.GetUtf16Bytes(0, aEntryMetadata.DataLength / 2);
+                xValue = Encoding.Unicode.GetBytes(aValue.ToCharArray(), 0, (int)aEntryMetadata.DataLength / 2);
 
```

Comment:
As ``aValue`` is a C# string already the conversion using ToCharArray() is not needed, as the last parameter is charCount I think the division per 2 is not needed anymore.